pkg/controller/sparkapplication/controller.go

```diff
@@ -320,14 +320,12 @@ func (c *Controller) getAndUpdateDriverState(app *v1beta1.SparkApplication) erro
 		}
 		app.Status.AppState.State = newState
 	} else {
-		glog.Warningf("driver not found for SparkApplication: %s/%s", app.Namespace, app.Name)
-		// The application has not terminated and has a recorded driver Pod, but no driver Pod was found for it.
-		// This is likely because the driver Pod was deleted. In this case, set the application state to FailingState.
-		if app.Status.TerminationTime.IsZero() && app.Status.DriverInfo.PodName != "" {
-			app.Status.AppState.ErrorMessage = "Driver Pod not found"
-			app.Status.AppState.State = v1beta1.FailingState
-			app.Status.TerminationTime = metav1.Now()
-		}
+		glog.Warningf("driver not found for SparkApplication: %s/%s. Marking SparkApplication as Failing.", app.Namespace, app.Name)
```

Comment:
This will cause issue for applications that have been submitted but have not started running (no driver pod has been created).

Reply:
This is still an issue even if you try fetching the driver pod from the API server. The driver pod may have not been created yet.